packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.t.sol

```diff
@@ -17,11 +18,55 @@ contract ConstantFlowAgreementV1IntegrationTest is FoundrySuperfluidTester {
 
     function testBobAliceLoop(int96 flowRate) public {
         _helperCreateFlow(superToken, alice, bob, flowRate);
-        
+
         _warpAndAssertAll(superToken);
 
         _helperCreateFlow(superToken, bob, alice, flowRate);
 
         _warpAndAssertAll(superToken);
     }
+
+    // there should be no gas limit which causes the NFT hook to fail with the tx succeeding
+    function testNFTHookOutOfGasRevertsWholeTx(uint256 gasLimit) public {
```

Comment:
did you observe any failure without the fix with such fuzzing?

Reply:
yes, reliably failed.
Example:
```
[FAIL. Reason: CFA_NFT_INVALID_TOKEN_ID(); counterexample: calldata=0xa34c991a00000000000000000000000000000000000000000000000000000000000060c4 args=[24772 [2.477e4]]] testNFTHookOutOfGasRevertsWholeTx(uint256) (runs: 27, μ: 415430, ~: 419640)
Logs:
  Bound Result 424774

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 55.35ms
```